### PR TITLE
[HKET] Add Extractor

### PR DESCRIPTION
**Pull request by Ashish0804**
### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

closes https://github.com/yt-dlp/yt-dlp/issues/4173

### CODE REVIEW

File: yt_dlp/extractor/hket.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        },
+    }]
+
+    def generate_m3u8(self, vid, video_filename):
+        # In case this extractor is broken, it's likely that these values have changed.
+        # These were found in video-autoplay-<whatever>.min.js with same keys.
+
+        domain = 'd3okrhbhk6fs3y.cloudfront.net'
+        source_dir = 'p1prod'
+        r = int(5e3 * (int(vid / 5e3) | 0))
```
[the remark this answers is not in the document]
Comment (Ashish0804) [reply yt_dlp/extractor/hket.py:46]: > Why am I requested for review?!



> Why am I requested for review?!

pukkandan seemed bust and since u r the other active maintainer, i added u aswell :)

[the remark this answers is not in the document]
Comment (Ashish0804) [reply yt_dlp/extractor/hket.py:46]: > 5e3 * (vid // 5e3)

This code was copied from js. 

I do not have any urls to test if removing int and the binary OR will result in any errors or not, are u sure there are no edge cases and the two formulas are extactly same?

File: yt_dlp/extractor/hket.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        return f'https://{domain}/{source_dir}/{r}/{vid}/{video_filename}.m3u8'
+
+    def find_data_property(self, property, webpage, validation=str, fatal=True):
+        try:
+            return validation(self._search_regex(fr'data-{property}\s*=\s*"([^"]+)"', webpage, property, fatal=fatal))
+        except Exception as e:
+            if fatal:
+                return e
```
[the remark this answers is not in the document]
Comment (Ashish0804) [reply yt_dlp/extractor/hket.py:55]: I just want this line to throw the error.

should it be `return e` or `throw e` or `raise e` or anything else?

File: yt_dlp/extractor/hket.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lesmiscore) [reply yt_dlp/extractor/hket.py:46]: > pukkandan seemed bust and since u r the other active maintainer, i added u aswell :)

Okay, why not

> I do not have any urls to test if removing int and the binary OR will result in any errors or not, are u sure there are no edge cases and the two formulas are extactly same?

In JavaScript, `value | 0` casts `value` into an integer (In python `12.6|0` throws error!)
Since the result of division is passed to `| 0`, `(vid / 5e3) | 0` is semantically equal to `vid // 5e3`

File: yt_dlp/extractor/hket.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lesmiscore) [reply yt_dlp/extractor/hket.py:55]: Looking for how `find_data_property` is used it should be `raise`...
